cloud/docker.go

```diff
@@ -110,10 +110,20 @@ func (m *dockerManager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host
 		return nil, errors.Wrapf(err, "Invalid Docker settings in distro '%s'", h.Distro.Id)
 	}
 
+	parent, err := distro.FindOne(distro.ById(h.ParentID))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error finding parent of host %s", h.Id)
+	}
+
+	hostIP, err := parent.GetHostIP()
```

Comment:
This is getting the IP address from the distro object, which still assumes that all parent hosts with the same distro connect to the same machine. Shouldn't this hostIP be associated with the hosts themselves, rather with distros?

Reply:
Fixed to get the `HostIP` from the parent's host document.